test/unit/gateways/card_stream_test.rb

```diff
@@ -311,6 +322,47 @@ def test_default_3dsecure_required
     end.respond_with(successful_purchase_response)
   end
 
+  def test_3ds2_data
+    stub_comms do
+      @gateway.purchase(142, @visacreditcard, @visacredit_options.merge(@visacredit_three_ds_options))
+    end.check_request do |_endpoint, data, _headers|
```

Comment:
💬  Comment:

Here you can just pass `check_request(skip_response: true)` because you are not interested in testing how the adapter is handling the response, you are just asserting how the 3ds data gets formatted.

with that you can remote the last part of line 335 and totally remote the uneeded `successful_purchase_response_with_3dsecure` method.